src/output-json.c

```diff
@@ -742,14 +743,49 @@ static int MacSetIterateToJSON(uint8_t *val, MacSetSide side, void *data)
     return 0;
 }
 
-static int CreateJSONEther(JsonBuilder *js, const Packet *p, const Flow *f)
+static int CreateJSONEther(
+        JsonBuilder *js, const Packet *p, const Flow *f, enum OutputJsonLogDirection dir)
 {
     if (p != NULL) {
         /* this is a packet context, so we need to add scalar fields */
         if (p->ethh != NULL) {
             jb_open_object(js, "ether");
-            uint8_t *src = p->ethh->eth_src;
-            uint8_t *dst = p->ethh->eth_dst;
+            uint8_t *src;
+            uint8_t *dst;
+            switch (dir) {
+                case LOG_DIR_FLOW:
```

Comment:
`LOG_DIR_FLOW` and `LOG_DIR_FLOW_TOSERVER` behave the same way ? What is the difference ?

Reply:
I don't think there is a real difference there. LOG_DIR_FLOW is used by the flow logger. And LOG_DIR_FLOW_TOSERVER is used in netflow for the to server way. Maybe we can merge them at least in this switch.